App.tsx

```diff
@@ -22,184 +30,44 @@ const theme = {
-    const { $isDataReady, $stored, $currentName, $currentItems } = gutkasFetched;
-    this.setState({ isDataReady: $isDataReady, gutkas: $stored, currentName: $currentName, currentItems: $currentItems });
+const App = () => {
+  const gutkaApi = useApi(gutkaAPIFactory, initialGutkaState);
+  const viewerApi = useApi(viewerApiFactory, initialViewerState);
+  const searchApi = useApi(searchApiFactory, initalSearchState);
 
-    const settingsFetched = await fetchSettings();
-    const { $displayEngTransl, $displayPunTansl, $displayTranslit, $gurmukhiSize, $translSize, $translitSize } = settingsFetched;
-    this.setState({ displayEngTransl: $displayEngTransl, displayPunTansl: $displayPunTansl, displayTranslit: $displayTranslit, gurmukhiSize: $gurmukhiSize, translSize: $translSize, translitSize: $translitSize });
-  }
-  toggleModal = () => {
-    this.setState(prevState => ({ modalVisible: !prevState.modalVisible }))
-  }
-  updateSearchType = (type: SearchType) => { this.setState({ searchType: type }) }
-  updateQueryType = (type: QueryType) => this.setState({ queryType: type })
-  toggleEditMode = () => { this.setState((prevState) => ({ isEditMode: !prevState.isEditMode })); }
-
-  updateCurrentGutka = (newGutka: string) => {
-    const gutka = findCurrentGutka(this.state.gutkas, newGutka);
-    this.setState({ currentName: newGutka, currentItems: getGutkaItems(gutka) });
-  }
-
-  updateFontSize = (element: string, size: number) => {
-    const fontSetting = `${element}Size`;
-    this.setState(prevState => ({
-      ...prevState,
-      [fontSetting]: size,
-    }));
-  }
-  updateDisplay = (element: string, value: boolean) => {
-    const displaySetting = `display${element}`;
-    this.setState(prevState => ({
-      ...prevState,
-      [displaySetting]: value,
-    }));
-  }
-  removeFromGutka = (id: number) => {
-    const { gutkas, currentName } = this.state;
-    const indexOf = findCurrentGutkaIndex(gutkas, currentName);
-    const entryIndex = findEntry(gutkas[indexOf], id);
-    gutkas[indexOf].items.splice(entryIndex, 1);
-    this.setState({ gutkas: gutkas, currentItems: _.values(gutkas[indexOf].items) });
-    saveGutkas(gutkas);
-  }
-  addToGutka = (entryid: number, mainLine: string, entrytype: gutkaEntry) => {
-    const newEntry: entryObj = {
-      id: entryid,
-      mainLine: mainLine,
-      type: entrytype
-    }
-    const { gutkas, currentName } = this.state;
-    const allGutkas = gutkas;
-    const indexOfCurrent = findCurrentGutkaIndex(allGutkas, currentName);
-
-    allGutkas[indexOfCurrent].items = [...allGutkas[indexOfCurrent].items, newEntry];
-
-    this.setState({ gutkas: allGutkas, currentItems: _.values(allGutkas[indexOfCurrent].items) });
-    saveGutkas(allGutkas);
-  }
-  createGutka = (gutkaName: string) => {
-    const newGutka: storedGutka = {
-      name: gutkaName,
-      items: [],
+  useEffect(() => {
+    const checkDB = async () => {
+      NetInfo.fetch().then(async state => {
```

Comment:
better would be to extract this to an async function and then use function`.then()` in useEffect

Reply:
also refactor netinfo usage
https://github.com/react-native-community/react-native-netinfo/blob/master/README.md#netinfostate
https://github.com/react-native-community/react-native-netinfo#usenetinfo